imports/ui/components/user_mgmt/user_mgmt.js

```diff
@@ -24,14 +26,15 @@ Template.userMgmt.helpers({
     return UsersIndex;
   },
   count: function() {
-    return UsersIndex.getComponentDict().get('count');
+    return Meteor.users.find({}).length;
   },
 });
 
 Template.userMgmt.events({
   'click input[type="checkbox"]': function(e) {
-    var profile = Meteor.users.findOne({username: $(e.target).parent().prev().html()}).profile;
-    profile["banned"] = !profile["banned"];
-    Meteor.users.update(Meteor.users.findOne({username: $(e.target).parent().prev().html()})._id, {$set: {profile: profile}});
+    var id = Meteor.users.findOne({username: $(e.target).parent().prev().html()})._id;
+    var profile = Profiles.findOne({userId: id});
+    var newVal = !profile["banned"];
+    Profiles.update(profile._id, {$set: {banned: newVal}});
```

Comment:
Couple questions spring to mind: 

1. We need to make sure that moderators cannot ban admins. Is that setup in our permissions already?
2. Also need to ensure that regular users cannot access this page or edit others' profiles, though I believe you've already done this.